eng/Directory.Build.Data.props

```diff
@@ -112,7 +112,7 @@
     <DelaySign>false</DelaySign>
     <PublicSign>false</PublicSign>
     <ImportDefaultReferences>false</ImportDefaultReferences>
-    <UseProjectReferenceToAzureCore>false</UseProjectReferenceToAzureCore>
+    <UseProjectReferenceToAzureCore>true</UseProjectReferenceToAzureCore>
```

Comment:
Can we avoid changing this until we actually have some library that depends on the new changes? I want to start getting into the model of using a package dependency which is what will happen after GA.

Reply:
There are already multiple changes that use project reference https://github.com/Azure/azure-sdk-for-net/pull/7957 being the largest. I included this switch in this PR to test that nothing accidentally breaks.

Reply:
Ok.... wow that is a lot of name changes :(